### PR TITLE
ICMSLST-1285 Switch to using Django's EmailMessage class

### PR DESCRIPTION
**Pull request by davidwtbuxton**
Instead of using the MailServer class for both POP3 and SMTP, use Django's
mail backends for doing the SMTP stuff. This involves removing SMTP from
MailServer, adding a helper to get a Django mail backend configured with
different SMTP settings, and constructing a Django EmailMessage instance
instead of a plain Python message instance.

Then when sending email, we need to get the appropriate Django mail backend
and use that to send the Django EmailMessage.

When running tests Django creates a locmem SMTP backend which never sends
nothing, but we can access django.core.mail.outbox to see what is "sent".
Mailhog is not required to run the tests (but is still super useful for
local development).

To simplify setup, several project settings have been made optional, with
default values that allow the application to start, which will then need
setting when deployed to production.

### CODE REVIEW

File: mail/servers.py [hunk 1]
```diff
@@ -1,8 +1,35 @@
 import logging
 import poplib
-import smtplib
 
 from django.conf import settings
+from django.core import mail
+
+
+def get_smtp_connection(name: str = None):
```
Comment (davidwtbuxton) [inline mail/servers.py:8]: This is the weirdest part of these changes in my opinion. There are 3 places which use the `get_smtp_connection` function, and they only use the default (None), "spire" and "mock" values. So we don't use the "incoming" and "dit" variations. And I think for ICMS' needs we won't ever want to send/relay mail on anything except the default SMTP server (and maybe a test/mock SMTP server).

File: mail/tests/test_send_lite_licence_updates_task.py [hunk 2]
```diff
@@ -38,6 +38,7 @@ def test_reply_received(self, send):
         send_licence_data_to_hmrc.now()
         self.assertEqual(LicencePayload.objects.filter(is_processed=True).count(), 0)
 
+    @skip("Fails, also fails for lite-hmrc project")
```
Comment (davidwtbuxton) [inline mail/tests/test_send_lite_licence_updates_task.py:41]: This test is broken. But it is also broken in lite-hmrc, except the CI/CD test runner ignored any test with the "end-to-end" tag, so no-one had noticed.

File: mail/libraries/builders.py [hunk 3]
```diff
@@ -156,39 +156,30 @@ def build_licence_data_file(licences, run_number) -> (str, str):
...
-    multipart_msg["From"] = settings.EMAIL_USER  # the SMTP server only allows sending as itself
-    multipart_msg["To"] = email_message_dto.receiver
-    multipart_msg["Subject"] = email_message_dto.subject
-    multipart_msg["name"] = email_message_dto.subject
-    multipart_msg.attach(MIMEText("\n\n", "plain", "iso-8859-1"))
-    payload = MIMEApplication(file)
-    payload.set_payload(file)
-    payload.add_header(
-        "Content-Disposition", f'attachment; filename="{email_message_dto.attachment[0]}"',
+    part1 = MIMEText("\n\n", _charset="iso-8859-1")
```
Comment (davidwtbuxton) [inline mail/libraries/builders.py:170]: I have no idea what purpose is served by this empty MIME part. Smells very unnecessary.

File: mail/libraries/builders.py [hunk 4]
```diff
@@ -156,39 +156,30 @@ def build_licence_data_file(licences, run_number) -> (str, str):
...
+    part2.add_header("Content-Disposition", "attachment", filename=filename)
+
+    message = mail.EmailMessage(
+        subject=email_message_dto.subject,
+        body="",
+        from_email=settings.EMAIL_USER,  # the SMTP server only allows sending as itself
+        to=[email_message_dto.receiver],
+        attachments=[part1, part2],
     )
-    payload.add_header("Content-Transfer-Encoding", "7bit")
```
Comment (davidwtbuxton) [inline mail/libraries/builders.py:187]: I don't think this code was doing what was intended. The generated message included the default `Content-transfer-encoding: base64` header and then this additional header.

File: mail/servers.py (hunk 1 above)
Comment (MattHolmes123) [reply mail/servers.py:8]: either an enum or type `name` as a `Literal`

**Review by MattHolmes123 — APPROVED** [review]

File: mail/servers.py (hunk 1 above)
Comment (davidwtbuxton) [reply mail/servers.py]: Updated to use an enum.

**Pull request merged**